pay-api/tests/unit/api/test_eft_short_name_history.py

```diff
@@ -0,0 +1,110 @@
+                                              auth_account_id='1234', name='ABC-BRANCH', branch_name='BRANCH').save()
+    short_name = factory_eft_shortname(short_name='TESTSHORTNAME1').save()
+
+    EFTHistoryService.create_funds_received(EFTHistory(short_name_id=short_name.id,
+                                                       amount=351.50,
+                                                       credit_balance=351.50)).save()
+
+    EFTHistoryService.create_statement_paid(EFTHistory(short_name_id=short_name.id,
+                                                       amount=351.50,
+                                                       credit_balance=0,
+                                                       payment_account_id=payment_account.id,
+                                                       related_group_link_id=1,
+                                                       statement_number=1234)).save()
+
+    EFTHistoryService.create_statement_reverse(EFTHistory(short_name_id=short_name.id,
+                                                          amount=351.50,
+                                                          credit_balance=351.50,
+                                                          payment_account_id=payment_account.id,
+                                                          related_group_link_id=2,
+                                                          statement_number=1234)).save()
+
+    return payment_account, short_name
+
+
+def test_search_short_name_history(session, client, jwt, app):
+    """Assert that EFT short names history can be searched."""
+    token = jwt.create_jwt(get_claims(roles=[Role.MANAGE_EFT.value]), token_header)
+    headers = {'Authorization': f'Bearer {token}', 'content-type': 'application/json'}
+    payment_account, short_name = setup_test_data()
+    rv = client.get(f'/api/v1/eft-shortnames/{short_name.id}/history', headers=headers)
+    assert rv.status_code == 200
+
+    result_dict = rv.json
+    assert result_dict is not None
+    assert result_dict['page'] == 1
+    assert result_dict['total'] == 3
+    assert result_dict['limit'] == 10
+    assert result_dict['items'] is not None
+    assert len(result_dict['items']) == 3
+
+    transaction_date = EFTHistoryService.transaction_date_now().strftime('%Y-%m-%d')
+    funds_received = result_dict['items'][2]
+    assert not funds_received['isReversible'] is True
+    assert funds_received['accountBranch'] is None
+    assert funds_received['accountId'] is None
+    assert funds_received['accountName'] is None
+    assert funds_received['amount'] == 351.50
+    assert funds_received['historicalId'] is not None
+    assert funds_received['shortNameBalance'] == 351.50
+    assert funds_received['shortNameId'] == short_name.id
+    assert funds_received['statementNumber'] is None
+    assert funds_received['transactionType'] == EFTHistoricalTypes.FUNDS_RECEIVED.value
```

Comment:
Probably worth using the stacking style for the unit test:

![image](https://github.com/user-attachments/assets/5c4bab26-021c-44c5-8476-56f1a262acec)

Reply:
alternatively.. could separate the tests.. It just gets hard to debug like the search unit tests when it's long

Reply:
Will take a look to see what make sense

Reply:
- Parameterized the statement related history as the checks were aligned.
- Split out funds received history as it is a bit different in relation to account information expected
- added in a test for specifically getting is_reversible to be True